File: iosApp/iosApp/ProfileSetup/ProfileSetupStepScaffold.swift
import SwiftUI

/// The title+subtitle block every step shares, wrapping that step's own unique body -
/// mirrors Android's `ProfileSetupStepScaffold.kt`.
struct ProfileSetupStepScaffold<Content: View>: View {
    let title: String
    let subtitle: String
    @ViewBuilder let content: () -> Content

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            Text(title)
                .font(CyraFont.headlineMedium())
                .foregroundColor(.cyraOnSurface)
            Spacer().frame(height: 8)
            Text(subtitle)
                .font(CyraFont.bodyMedium())
                .foregroundColor(.cyraOnSurfaceVariant)
            Spacer().frame(height: 24)
            content()
        }
    }
}

File: androidApp/src/main/kotlin/subha/app/cyra/ui/profilesetup/ProfileSetupStepScaffold.kt
package subha.app.cyra.ui.profilesetup

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.height
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp

/**
 * The title+subtitle block every step shares, wrapping that step's own unique body -
 * keeps each `ProfileSetup*Step` composable down to just its distinctive fields, not a
 * copy of this boilerplate 8 times over.
 */
@Composable
fun ProfileSetupStepScaffold(
    title: String,
    subtitle: String,
    modifier: Modifier = Modifier,
    content: @Composable () -> Unit,
) {
    Column(modifier = modifier) {
        Text(text = title, style = MaterialTheme.typography.headlineMedium, color = MaterialTheme.colorScheme.onSurface)
        Spacer(modifier = Modifier.height(8.dp))
        Text(text = subtitle, style = MaterialTheme.typography.bodyMedium, color = MaterialTheme.colorScheme.onSurfaceVariant)
        Spacer(modifier = Modifier.height(24.dp))
        content()
    }
}
